payloads/v8/channel.ts

```diff
@@ -927,85 +931,107 @@ export const enum ComponentType {
 }
 
 /**
- * https://discord.com/developers/docs/interactions/message-components#component-object
+ * https://discord.com/developers/docs/interactions/message-components#action-rows
  */
-export interface APIActionRowComponent extends APIBaseComponent {
-	/**
-	 * The type of the component
-	 */
-	type: ComponentType.ActionRow;
+export interface APIActionRowComponent extends APIBaseMessageComponent<ComponentType.ActionRow> {
 	/**
 	 * The components in the ActionRow
 	 */
 	components: Exclude<APIMessageComponent, APIActionRowComponent>[];
 }
 
 /**
- * https://discord.com/developers/docs/interactions/message-components#buttons-button-object
+ * https://discord.com/developers/docs/interactions/message-components#buttons
  */
-export interface APIButtonComponent extends APIBaseComponent {
-	/**
-	 * The type of the component
-	 */
-	type: ComponentType.Button;
+interface APIButtonComponentBase<Style extends ButtonStyle> extends APIBaseMessageComponent<ComponentType.Button> {
 	/**
 	 * The label to be displayed on the button
 	 */
 	label?: string;
-	/**
-	 * The custom_id to be sent in the interaction when clicked
-	 */
-	custom_id?: string;
+
```

Comment:
I only looked at a few files, but it doesn't seem to look like there were empty lines like these.
```suggestion
```